### ISSUE TITLE
please make barChart also available horizontally

### ISSUE THREAD

**Issue by d2a-raudenaerde** [dc-js/dc.js#693]
I know it becomes more of a 'rowChart' then, but the functionality of both charts is different (label placement, etc), and I need the bars to go from left-to-right.

**Comment by nordfjord** [comment]
Any progress on at least getting labels on the left side? I find it incredibly frustrating to try to read text when there's no bar behind it, it is in fact impossible to read white on white.

**Comment by gordonwoodhull** [comment]
@nordfjord, please submit a patch. I think the workaround most people use is to reverse the color of the text.

**Comment by nordfjord** [comment]
@gordonwoodhull I haven't familiarized myself with your source, but I'll get on it when I have time, and will try to find a working way to provide reasonable label positioning by default.
similar to
![bar chart](https://upload.wikimedia.org/wikipedia/commons/3/35/Human_losses_of_world_war_two_by_country.png)

In my opinion the bar chart , or in the case of dc  the 'row chart', was perfected when it was invented by William Playfair, and we have only found ways to make it worse since. You can see the original bar chart [here](https://en.wikipedia.org/wiki/Bar_chart#/media/File:Playfair_Barchart.gif)

**Comment by gordonwoodhull** [comment]
Heh, too true. Hand-design does have the advantage of customizability, if time consuming.

I think the tricky thing here is to adjust the margin to fit the text. If you don't worry about that and just set the margin manually, then I think it's just a matter of right-aligning the text against the Y axis instead of left-aligning it. (dc.js mostly does not set the margins automatically, a frequent complaint.)

**Comment by nordfjord** [comment]
Indeed, and it gets especially tricky once you have N rows. Are there any plans to support something like a minimap, so that you could reasonably have N rows and then simply scroll, in stead of reducing the height of bars below 1px, and therefore being completely useless?
